distutils/command/build_scripts.py

```diff
@@ -106,18 +105,8 @@ def _copy_script(self, script, outfiles, updated_files):  # noqa: C901
         if shebang_match:
             log.info("copying and adjusting %s -> %s", script, self.build_dir)
             if not self.dry_run:
-                if not sysconfig.python_build:
-                    executable = self.executable
-                else:
-                    executable = os.path.join(
-                        sysconfig.get_config_var("BINDIR"),
-                        "python{}{}".format(
-                            sysconfig.get_config_var("VERSION"),
-                            sysconfig.get_config_var("EXE"),
-                        ),
-                    )
                 post_interp = shebang_match.group(1) or ''
-                shebang = "#!" + executable + post_interp + "\n"
+                shebang = f"#!python{post_interp}\n"
```

Comment:
I believe this is the wrong approach. This code belongs to the `build_scripts` code, which may be invoked via the CLI, and even receives a `executable` argument, which is now explicitly ignored.

The `#!python` functionality is part of [PEP 427](https://peps.python.org/pep-0427), and as such, should be implemented in the wheel building code, not here.

Reply:
It seems `bdist_wheel` already implements this functionality itself, so the `build_editable` issue seems like a bug on the `setuptools.build_meta` backend. I believe fixing it should be as simple as setting `executable` there to `python`, just like `bdist_wheel` does.

https://github.com/pypa/wheel/blob/c8f1f09d1abee526851b4e058331db5489a335c4/src/wheel/_bdist_wheel.py#L382-L384